DOCSP-48120: elemmatch method overload (#537)
mongodb/docs-csharp: source/fundamentals/builders.txt

   var builder = Builders<Flower>.Filter;
   var filter = builder.ElemMatch(f => f.Season, s => s == "Summer");

.. tip:: ElemMatch() Overload

>    The ``ElemMatch()`` method has an overload to accept a single filter

Reviewer: ```suggestion
   The ``ElemMatch()`` method has an overload that accepts a single filter
```

>    parameter. This functionality supports performing queries that
>    include nested ``$elemMatch`` statements.

Reviewer: S: for simplicity

```suggestion
   parameter. You can use this overload to perform queries that
   include nested ``$elemMatch`` statements.
```

>    include nested ``$elemMatch`` statements.

Reviewer 2: While true that is not the primary purpose of this overload.

The primary purpose is to let you write ElemMatch queries against the values themselves instead of against a property of the values.

Author: Updated the text
